### PR TITLE
fix(schema): exclude functions from `DeepPartial`

### LINKED ISSUE #5477: `nuxt.config`'s `vue` types are incorrect
### Environment

------------------------------
- Operating System: `Windows_NT`
- Node Version:     `v16.13.1`
- Nuxt Version:     `3.0.0-rc.4`
- Package Manager:  `npm@8.1.2`
- Builder:          `vite`
- User Config:      `modules`, `css`, `typescript`, `build`, `runtimeConfig`, `vue`
- Runtime Modules:  `@nuxtjs/tailwindcss@5.1.2`, `floating-vue/nuxt`, `@pinia/nuxt@0.1.9`, `@vueuse/nuxt@8.6.0`
- Build Modules:    `-`
------------------------------

### Reproduction

[StackBlitz](https://stackblitz.com/edit/github-pc4kbz?file=nuxt.config.ts&view=editor)

### Describe the bug

Types do not seem to be correct for the `vue` property in the `nuxt.config` file.

image

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves #5477

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

An early return for functions solves some issues with functional types (including missing types for `hook`, for example). There were also a couple of incorrect types in `_app`.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/schema/src/config/_app.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export default {
      * Properties that will be set directly on `Vue.config` for vue@2.
      *
      * @see [vue@2 Documentation](https://v2.vuejs.org/v2/api/#Global-Config)
-     * @type {import('vue/types/vue').VueConfiguration}
+     * @type {typeof import('vue/types/vue').VueConfiguration}
```
Comment (pi0) [inline packages/schema/src/config/_app.ts:16]: Does it makes sense to automatically infer this in untyped?

**Review by pi0 — APPROVED** [review]

File: packages/schema/src/config/_app.ts (hunk 1 above)
Comment (danielroe) [reply packages/schema/src/config/_app.ts:16]: The issue is that we have a two stage build here.

Comment (pi0) [reply packages/schema/src/config/_app.ts:16]: Sure. I mean if untyped can add `typeof` prefix to `import` in `@type` annotations.

Comment (pi0) [reply packages/schema/src/config/_app.ts:16]: (anyway, this fix makes sense. Was thinking of a way to avoid future issues like this)

**Pull request merged**